### ISSUE TITLE
Test failure in CI build 10024

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#3379]
The following test appears to have failed:

[#10024](https://circleci.com/gh/cockroachdb/cockroach/10024):

```
W1210 01:07:01.517733 586 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
W1210 01:07:01.794612 586 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
I1210 01:07:01.794732 586 client/db.go:483  failed batch: failed to send RPC request Server.Batch: client is unhealthy
W1210 01:07:01.811473 586 rpc/client.go:249  unexpected EOF
I1210 01:07:01.811847 586 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestAuthentication (0.37s)
	db_test.go:258: failed to send RPC request Server.Batch: client is unhealthy
=== RUN   TestRangeLookupWithOpenTransaction
I1210 01:07:01.882081 586 rpc/clock_offset.go:154  monitoring cluster offset
I1210 01:07:01.890569 586 multiraft/multiraft.go:579  node 1 starting
I1210 01:07:01.892436 586 raft/raft.go:441  [group 1] 1 became follower at term 5
I1210 01:07:01.893719 586 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1210 01:07:01.893886 586 multiraft/multiraft.go:999  node 1 campaigning because initial confstate is [1]
I1210 01:07:01.893996 586 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1210 01:07:01.894111 586 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1210 01:07:01.894213 586 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6
--

I1210 01:07:08.798933 586 stopper.go:241  draining; tasks left:
1      storage/replica.go:1436
I1210 01:07:08.803223 586 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestPropagateTxnOnPushError (0.47s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	38.003s
=== RUN   TestHeartbeatSingleGroup
I1210 01:06:23.875977 559 multiraft/multiraft.go:579  node 1 starting
I1210 01:06:23.877046 559 multiraft/multiraft.go:579  node 2 starting
I1210 01:06:23.878088 559 raft/raft.go:441  [group 1] 1 became follower at term 5
I1210 01:06:23.878389 559 raft/raft.go:234  [group 1] newRaft 1 [peers: [1,2], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1210 01:06:23.879008 559 raft/raft.go:441  [group 1] 2 became follower at term 5
I1210 01:06:23.879187 559 raft/raft.go:234  [group 1] newRaft 2 [peers: [1,2], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1210 01:06:23.879520 559 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1210 01:06:23.879801 559 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1210 01:06:23.879993 559 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6

```
Please assign, take a look and update the issue accordingly.

**Comment by tschottdorf** [comment]
#2642? Though it's weird that we should've taken `5` attempts for talking to a local test server.

**Comment by tschottdorf** [comment]
I think @petermattis fixed this in #3421.
